binary/test/Test/Cardano/Binary/Serialization.hs

```diff
@@ -147,6 +152,26 @@ instance FromCBOR TestStruct where
       <*> fromCBOR
       <*> fromCBOR
       <*> fromCBOR
+      <*> fromCBOR
+
+genUTCTime :: Gen Time.UTCTime
+genUTCTime = Time.UTCTime
+  <$> genDay
+  <*> genDiffTimeOfDay
+  where
+  -- UTC time takes a DiffTime s.t. 0 <= t < 86401s
+  genDiffTimeOfDay :: Gen Time.DiffTime
+  genDiffTimeOfDay = Time.picosecondsToDiffTime <$>
+    Gen.integral (Range.constantFrom 0 0 ((86401e12) - 1))
+
+genDay :: Gen Time.Day
+genDay = Time.fromOrdinalDate <$> genYear <*> genDayOfYear
+
+genYear :: Gen Integer
+genYear = Gen.integral (Range.linear (-10000) 10000)
+
+genDayOfYear :: Gen Int
+genDayOfYear = Gen.int (Range.constantFrom 1 1 366)
```

Comment:
What happens in leap years?

Reply:
Yes, this is perhaps a bit subtle, but `fromOrdinalDate` should take care of clipping this in the case of non-leap years. I figured the random distribution having that small quirk was not worth fixing with extra complexity in the generator.